docs/sources/flow/reference/components/prometheus.exporter.postgres.md

```diff
@@ -0,0 +1,125 @@
+}
+```
+
+## Arguments
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`data_source_names`                  | `list(secret)`      | Specifies the Postgres server(s) to connect to.  |         | yes
+`disable_settings_metrics`           | `bool`              | Disables collection of metrics from pg_settings. | `false` | no
+`disable_default_metrics`            | `bool`              | When `true`, only exposes metrics supplied from `custom_queries_path`. | `false` | no
+`autodiscover_databases`             | `bool`              | Whether to automatically discover databases to scrape metrics from. | `false` | no
+`exclude_databases`                  | `list(string)`      | A list of databases to ignore when `autodiscover_databases` is `true` | | no
+`include_databases`                  | `list(string)`      | Includes only specific databases (excluding all others) when autodiscover_databases is `true` | | no
+`custom_queries_path`                | `string`            | Path to YAML file containing custom queries to expose as metrics. | | no
+
+The format for connection strings in `data_source_names` can be found in the [official postgresql documentation](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING).
+
+See examples for the `custom_queries_path` file in the [postgres_exporter repository](https://github.com/prometheus-community/postgres_exporter/blob/master/queries.yaml).
+
+**NOTE**: There are a number of environment variables that are not recommended for use, as they will affect _all_ `prometheus.exporter.postgres` components. A full list can be found in the [postgres_exporter repository](https://github.com/prometheus-community/postgres_exporter#environment-variables).
+
+## Blocks
+The `prometheus.exporter.postgres` component does not support any blocks, and is configured fully through arguments.
+
+## Exported fields
+The following fields are exported and can be referenced by other components:
+
+Name      | Type                | Description
+--------- | ------------------- | -----------
+`targets` | `list(map(string))` | The targets that can be used to collect `postgres` metrics.
+
+For example, `targets` can either be passed to a `prometheus.relabel`
+component to rewrite the metrics' label set, or to a `prometheus.scrape`
+component that collects the exposed metrics.
+
+## Component health
+
+`prometheus.exporter.postgres` is only reported as unhealthy if given
+an invalid configuration.
+
+## Debug information
+
+`prometheus.exporter.postgres` does not expose any component-specific
+debug information.
+
+## Debug metrics
+
+`prometheus.exporter.postgres` does not expose any component-specific
+debug metrics.
+
+## Examples
+
+This minimal example uses a `prometheus.exporter.postgres` component to collect metrics from a Postgres
+server running locally with all default settings:
```

Comment:
Are the defaults useful in this case? It seems like if autodiscover is off, it isn't going to collect very much. I've never used postgres exporter though, so I'm not sure.